app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', counter_cache: :post_counter
+  has_many :like
+  has_many :comment
+
+  def recent_5_comments
+    comment.order(created_at: :desc).limit(5)
+  end
+
+  def increment_post_count
+    author.increment!(:post_counter)
+  end
```

Comment:
- [ ] Optional: I would advise you to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html): `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private:
- Example:
```ruby
after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

Reply:
- [ ] Optional

As suggested by the previous reviewer, consider using rails active callback `after_save` for the method `update_posts_counter`. Doing this will save you a lot of headaches later since you won't have to call this method every time you create a post.